### DISCUSSION TITLE
Segmentation Fault when using union_by_name with read_parquet

### DISCUSSION THREAD

**Discussion by m-s-kistler** [duckdb/duckdb#6630] [Q&A]
I recently started exploring the use of union_by_name with the read_parquet() function, and came across a segmentation fault that is being thrown. If I remove union_by_name from the query, the segmentation fault does not occur, but I am not savvy enough with C++ to go down the rabbit hole and debug what is happening. Has anyone experience this before?

I broke down my complicated query into something simple that would still throw the segmentation fault, and toggling True/False with union_by_name is the only change I make that results in the error.

select 
 *
from 
 read_parquet(['path/to/parquet/my_parquet_files*.parquet'],union_by_name=True)

Any help would be appreciated...Thanks!

**Comment by Mytherin** [comment]
Thanks for the report - could you try to make a minimal example that includes the data that we need to reproduce the issue?

**Comment by m-s-kistler** [comment]
[segmentation_fault.zip](https://github.com/duckdb/duckdb/files/10932979/segmentation_fault.zip)

**Reply by m-s-kistler** [reply]
Added a sample set of python code and 2 parquet files that allowed me to reproduce the error. When testing, it seems that just adding `union_by_name` to the query causes the segmentation fault. It doesn't matter whether it is set to True or False.

**Reply by m-s-kistler** [reply]
I am using:
pandas v1.1.5
duckdb v0.7.1

**Reply by Mytherin** [reply]
Thanks! That's very helpful. I have reproduced the segfault and pushed a fix in #6642.

The problem is related to combining `union_by_name` together with the filter pushdown into the Parquet files.

**Reply by m-s-kistler** [reply]
Great! Appreciate the help.

**Reply by krzysztofzienkiewicz** [reply]
I think this might be back in the 1.1.0 release.

**Reply by krzysztofzienkiewicz** [reply]
Note: I'm using the azure extension and jdbc:
```
#
#  SIGSEGV (0xb) at pc=0x00007f3868f2ea03, pid=1300, tid=1732
#
# JRE version: OpenJDK Runtime Environment 18.9 (11.0.16+8) (build 11.0.16+8)
# Java VM: OpenJDK 64-Bit Server VM 18.9 (11.0.16+8, mixed mode, sharing, tiered, compressed oops, serial gc, linux-amd64)
# Problematic frame:
# C  [azure.duckdb_extension+0x8daa03]  duckdb::DefaultSecretGenerator::CreateDefaultEntryInternal(std::string const&)+0x2a3
```

**Reply by Mytherin** [reply]
Hard to say without any more information - but that stack trace suggests that there is a problem related to secret management with the Azure extension specifically, which is unrelated to this issue. Could you perhaps file an issue [there](https://github.com/duckdb/duckdb_azure/issues) with a reproducer, then we could take a look?
